### PR TITLE
[TECHNICAL SUPPORT] LPS-1521237,LPS-152212

### PR DESCRIPTION
**Pull request by NemethNorbert**
Hey,

[LPS-151237](https://issues.liferay.com/browse/LPS-151237)
[LPS-152212](https://issues.liferay.com/browse/LPS-152212)

Previous PR with only LPS-151237 : https://github.com/liferay-forms/liferay-portal/pull/1610

I decided to create one PR with the two issues, as they are both connected and LPS-152212 requires the changes from LPS-151237 as well.
With this PR the user will be able to save an empty value for a given locale and also change it back to the default locale's value.

Please review my changes

### CODE REVIEW

**Comment by NemethNorbert** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js [hunk 1]
```diff
@@ -4085,7 +4085,7 @@ AUI.add(
 							var defaultLocale = field.getDefaultLocale();
 
 							availableLanguageIds.forEach((locale) => {
-								if (!localizationMap[locale]) {
+								if (localizationMap[locale] === null) {
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js:4088]: @NemethNorbert could you please explain why it has to be `null` but not `undefined` in this condition?

Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js:4088]: Hi @NemethNorbert, 
On 7.3.x is actually `localizationMap[locale] == null`, which means that the condition will be true if `localizationMap[locale]` is `null` or `undefined`. 
But in this case, the condition will only be true if `localizationMap[locale]` is `null`.

Comment (NemethNorbert) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js:4088]: Hey @carolmariaabb,

you are right, I changed the `null` to `undefined` as the SF does not allow the usage of `==`
I pushed the change.

Thanks,

**Comment by NemethNorbert** [comment]
ci:test:sf

**Comment by NemethNorbert** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js [hunk 2]
```diff
@@ -1297,7 +1297,7 @@ AUI.add(
 							locale === defaultLocale ||
 							(localizationMap[defaultLocale] !== undefined &&
 								value !== localizationMap[defaultLocale]) ||
-							localizationMap[locale]
+							localizationMap[locale] !== undefined
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js:1300]: @NemethNorbert do you think it is necessary to check that `localizationMap[locale] !== null`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js [hunk 3]
```diff
@@ -4085,7 +4085,7 @@ AUI.add(
 							var defaultLocale = field.getDefaultLocale();
 
 							availableLanguageIds.forEach((locale) => {
-								if (!localizationMap[locale]) {
+								if (localizationMap[locale] === undefined) {
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js:4088]: @NemethNorbert do you think it is necessary to check that `localizationMap[locale] === null`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js (hunk 2 above)
Comment (NemethNorbert) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js:1300]: Hey @carolmariaabb , 
Yes that check allows to update the locale in the localisationMap even if its current value an empty string

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**